### PR TITLE
Aggregate perf metrics

### PR DESCRIPTION
**Pull request by katarzyna-z**
Signed-off-by: Katarzyna Kujawa <katarzyna.kujawa@intel.com>

I want to test it more and add more unit tests.
Related to #2608

### CODE REVIEW

**Comment by katarzyna-z** [comment]
Tested with 8 cores, 60 containers, 5 perf events. configuration file:
```
{
  "core": {
    "events": [
      ["instructions"],
      ["instructions_retired"],
      ["cycles"],
      ["ref-cycles"],
      ["context-switches"]
    ],
    "custom_events": [
      {
        "type": 4,
        "config": [
          "0x5300c0"
        ],
        "name": "instructions_retired"
      }
    ]
  }
}

```

**Aggregated perf events**
command: `sudo ./cadvisor -perf_events_config=perf/testing/perf.json -disable_metrics="accelerator,disk,diskIO,network,process,hugetlb,resctrl,udp,advtcp,referenced_memory,cpu_topology,tcp,sched,percpu"`
response time on /metrics endpoint: 0.419s
data size of response: 5MB

**Per CPU perf events**
command: `sudo ./cadvisor -perf_events_config=perf/testing/perf.json -disable_metrics="accelerator,disk,diskIO,network,process,hugetlb,resctrl,udp,advtcp,referenced_memory,cpu_topology,tcp,sched"`
response time on /metrics endpoint: 0.514s
data size of response: 14MB


@wacuuu Could you test this pull request in your environment?

**Comment by wacuuu** [comment]
baseline for disable_metrics is set to
`udp,referenced_memory,resctrl,hugetlb,cpu_topology,tcp,advtcp,sched,process`, i only add or remove `percpu` parameter

To measure the parameters i run the following command on the node that is running cadvisor in container:
`time curl localhost:8080/metrics > /dev/null`

Number of containers | Number of enabled perf counters | Time of scraping(s) | Scraped data volume(MB) | disable percpu
-- | -- | -- | -- | --
58 | 5 | 0.89s | 20M | yes
58 | 5 | 3.07s | 97M | no
-- | -- | -- | -- | --
58 | 10 | 1.82s | 21.8M | yes
58 | 10 | 12.92s |168M | no
-- | -- | -- | -- | --
32 | 10 | 0.86s | 15.3M | yes
32 | 10 | 4.81s | 21.4M | no


So the conclusion would be that `percpu` significantly improves the situation.

**Comment by katarzyna-z** [comment]
@dashpole it is ready for review.

File: docs/runtime_options.md [hunk 1]
```diff
@@ -140,6 +140,12 @@ cAdvisor stores the latest historical data in memory. How long of a history it s
 --perf_events_config="" Path to a JSON file containing configuration of perf events to measure. Empty value disables perf events measuring.
 ```
 
+Core perf events can be exposed on Prometheus endpoint per CPU or aggregated by event. It is controlled through `--disable_metrics` parameter with option `percpu`, e.g.:
+- `--disable_metrics="percpu"` - core perf events are aggregated
+- `--disable_metrics=""` - core perf events are exposed per CPU.
+
+Aggregated form of core perf events significantly decrease volume of data. For aggregated form of core perf events scaling ratio (`container_perf_metric_scaling ratio`) indicates average scaling ratio for specific event.
```
Comment (ppalucki) [inline docs/runtime_options.md:147]: IMO "average" is no useful comparing to the worst case when scaling happened
after aggregation with average equal 90% one doesnt know if the all cpus were scaled with by 10% or most of them were 100% and one was running only 5% and scaled 20 times (quite significant error)  

You should add or replace (avg) with min scaling ratio accross all cpus.

Comment (katarzyna-z) [reply docs/runtime_options.md:147]: ok so I'll replace average with min

**Comment by dashpole** [comment]
@katarzyna-z is it possible to unit test this?

File: metrics/prometheus_test.go [hunk 2]
```diff
@@ -44,6 +44,21 @@ func TestPrometheusCollector(t *testing.T) {
 	testPrometheusCollector(t, reg, "testdata/prometheus_metrics")
 }
 
+func TestPrometheusCollectorWithPerfAggregated(t *testing.T) {
```
Comment (katarzyna-z) [inline metrics/prometheus_test.go:47]: @dashpole Unit tests are available here. Do you think that I should add something?

**Pull request merged**